src/main/software/amazon/event/ruler/JsonRuleCompiler.java

```diff
@@ -401,25 +401,30 @@ private static Patterns processMatchExpression(final JsonParser parser) throws I
                     barf(parser, "Anything-But expression name not found");
                 }
                 final String anythingButObjectOp = parser.getCurrentName();
-                if (!Constants.PREFIX_MATCH.equals(anythingButObjectOp)) {
+                final boolean isPrefix = Constants.PREFIX_MATCH.equals(anythingButObjectOp);
+                final boolean isSuffix = Constants.SUFFIX_MATCH.equals(anythingButObjectOp);
+                if (!isPrefix && !isSuffix) {
                     barf(parser, "Unsupported anything-but pattern: " + anythingButObjectOp);
                 }
-                final JsonToken anythingButPrefix = parser.nextToken();
-                if (anythingButPrefix != JsonToken.VALUE_STRING) {
+                final JsonToken anythingButParamType = parser.nextToken();
+                if (anythingButParamType != JsonToken.VALUE_STRING) {
                     barf(parser, "prefix match pattern must be a string");
```

Comment:
We'd say "suffix" instead of "prefix" here.

Reply:
good catch.  this applies to both so i'll change it to prefix/suffix like the others?